FV3GFS/FV3GFS_io.F90

```diff
@@ -6979,6 +7003,30 @@ subroutine gfdl_diag_register(Time, Sfcprop, Gfs_diag, Model, Cldprop, Atm_block
       Diag(idx)%data(nb)%var2 => Sfcprop(nb)%uustar(:)
     enddo
 
+    ! KGao
```

Comment:
Are these so that we can have arrays holding the flux directly from the coupler? If so, why not use the arrays in the `Coupling%` data structure?

Reply:
Yes, we let the Sfcprop% data structure holds surface layer variables (heat fluxes, ocean temp, z0/zt) passed from the coupler. It is a convenient choice as some fields are already defined. I am not sure of the role of Coupling% in SHiELD physics. I don't think Joseph and I have touched it for the SHiELD-MOM6 coupling.

Reply:
That is fine; I'm not sure whether Coupling% is actually hooked up to anything. Sfcprop is already being used and is convenient, but be aware that it may be altered in subtle ways elsewhere in the physics driver.

Reply:
The coupling arrays may be useful if we need to accumulate something over timesteps instead of passing instantaneous values.

https://github.com/NOAA-GFDL/SHiELD_physics/blob/c93b388a6dfa6edad52e8eeb4a6b084428624c68/GFS_layer/GFS_typedefs.F90#L1911